src/common/lib/client/realtimechannel.ts

```diff
@@ -189,19 +189,20 @@ class RealtimeChannel extends Channel {
       return false;
     }
     if (options?.params) {
-      // Don't check against the `agent` param - it isn't returned in the ATTACHED message
-      const requestedParams = Object.assign({}, options.params);
-      delete requestedParams.agent;
-      if (Object.keys(requestedParams).length !== 0 && !this.params) {
+      // Don't check against the `agent` param
+      const requestedParams = omitAgent(options.params);
+      const existingParams = omitAgent(this.channelOptions.params);
```

Comment:
The problem here which is making tests fail is that in `RealtimeChannel.setOptions` we call the parent `Channel.setOptions` (which updates the value of `this.channelOptions`) before we call `_shouldReattachToSetOptions` so when updating options via `setOptions`, `existingParams` will always be equal to `requestedParams`.

Reply:
The simplest fix I can think of is to update `_shouldReattachToSetOptions` to take the existing options as an argument. That way you could copy the value of `this.ChannelOptions` before calling `Channel.prototype.setOptions` and pass them into `_shouldReattach...` explicitly

Reply:
thanks, fixed this